rust/extractor/src/translate/base.rs

```diff
@@ -626,17 +637,32 @@ impl<'a> Translator<'a> {
             if let Some(t) = type_refs
                 .next()
                 .and_then(ast::Type::cast)
-                .and_then(|t| self.emit_type(t))
+                .and_then(|t| self.emit_type(&t))
             {
                 generated::PathSegment::emit_type_repr(label, t, &mut self.trap.writer)
             }
             if let Some(t) = type_refs
                 .next()
                 .and_then(ast::PathType::cast)
-                .and_then(|t| self.emit_path_type(t))
+                .and_then(|t| self.emit_path_type(&t))
             {
                 generated::PathSegment::emit_trait_type_repr(label, t, &mut self.trap.writer)
             }
         }
     }
+
+    pub(crate) fn emit_item_expansion(&mut self, node: &ast::Item, label: Label<generated::Item>) {
+        (|| {
+            let semantics = self.semantics?;
+            let ExpandResult {
+                value: expanded, ..
+            } = semantics.expand_attr_macro(node)?;
+            // TODO emit err?
```

Comment:
It would be nice to register any failure reason as a diagnostic, so we can keep track of how many macro expansions were successful.

Reply:
after looking a bit at the rust analyzer code, I have the feeling the `err` in the `ExpandResult` gets recovered again in `parse_macro_expansion_error` which is called by our `emit_macro_expansion_parse_errors` function called just below. So I think we should ignore the `err` here, as otherwise we would be emitting a diagnostic twice about it.

Reply:
Ah that makes sense. I guess that leaves the `?` after the `cast` as a potential error that may be worth reporting. I suppose the result is always a `MacroItem`, but perhaps it could be just an `Item` if the macro expands to a single item, or an `Error` if something went wrong (although in that case I'd expect a parse error to be reported).